### PR TITLE
Make dialogs on windows non-sychronous.

### PR DESCRIPTION
**Pull request by jneem**
This does #1068 for windows.

### CODE REVIEW

File: druid-shell/src/platform/windows/window.rs [hunk 1]
```diff
@@ -1083,28 +1044,17 @@ impl WndProc for MyWndProc {
...
-            }
+            WM_CLOSE => self
+                .with_wnd_state(|s| s.handler.request_close())
+                .map(|_| 0),
             DS_REQUEST_DESTROY => {
                 unsafe {
                     DestroyWindow(hwnd);
                 }
                 Some(0)
             }
```
Comment (rhzk) [inline druid-shell/src/platform/windows/window.rs:1055]: If we make WindowHandle::close() deferred I think we could get rid of this custom message now,
unless there is some other reason we need it?

**Comment by jneem** [comment]
Now that mac support has landed, I've updated this PR to remove the backwards-compatibility glue.

**Comment by jneem** [comment]
Now that mac support has landed, I've updated this PR to remove the backwards-compatibility glue.

**Review by cmyr — APPROVED** [review]
Great. Looks like there might be some clippy stuff to clean up now, but happy to have this all sorted out, thanks again for taking this on!

**Pull request merged**